sql/sql_mode.go

```diff
@@ -30,8 +30,24 @@ const (
 	NoAutoValueOnZero    = "NO_AUTO_VALUE_ON_ZERO"
 	NoEngineSubstitution = "NO_ENGINE_SUBSTITUTION"
 	StrictTransTables    = "STRICT_TRANS_TABLES"
+	defaultSqlMode       = "no_engine_substitution,only_full_group_by,strict_trans_tables"
```

Comment:
(minor) You might consider exposing this constant (and uppercasing?) – then you could reference this default from `system_variables.go` instead of relying on those to match up. That way, if someone does change `system_variables.go` to change that value in the future, they'd have to come change it here and make sure this code gets updated, too. (you could also just reuse `defaultSqlMode` on line 47 if you want to avoid that extra `strings.Join` call.)